qiskit/algorithms/optimizers/qnspsa.py

```diff
@@ -228,9 +228,10 @@ def _point_sample(self, loss, x, eps, delta1, delta2):
         gradient_estimate = (loss_values[0] - loss_values[1]) / (2 * eps) * delta1
 
         # compute the preconditioner point estimate
+        fidelity_values = [float(f) for f in fidelity_values]
```

Comment:
Are the casts to float still required now that we explicitly use `diff = diff / thing` instead of `diff /= thing`?

Reply:
I think so because it failed again with `diff = diff / thing` with the same error